### PR TITLE
Filter Juniper mempools

### PR DESCRIPTION
**Pull request by TheMysteriousX**
The original mempool code ignored any memory value from a fan or sensor.

This adds skip_values for 'fan' and 'sensor' to the Junos YAML.

jnxOperatingMemory has also been substituted for jnxOperatingDramSize - jnxOperatingDramSize is deprecated as it is only 32 bits in size. 

fixes: #12435

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by TheMysteriousX** [comment]
This is a partial fix for #12435 - the wonky graph is still present.

The graphing issue is due to the mempools being stacked - this is valid on single entity systems, but on systems with e.g. multiple routing engines, that assumption does not hold.

**Comment by murrant** [comment]
Is there no test data for the fans and sensors?  Because no mempools were removed from the json.   If that is the case, updating/replacing one of the snmprecs would be great.

**Comment by TheMysteriousX** [comment]
Sure, I can rustle up some test data - it's only on some specific devices in specific configurations, looking at the existing snmprecs they're not devices that are affected, have been highly filtered, or have been captured with really old firmware.

I'll add some data for a large stack running a recent firmware.

File: includes/definitions/discovery/junos.yaml [hunk 1]
```diff
@@ -4,14 +4,23 @@ modules:
...
                         op: '='
                         value: 0
+                    -
+                        oid: JUNIPER-MIB::jnxOperatingDescr
+                        op: 'contains'
+                        value: 'Fan'
+                    -
+                        oid: JUNIPER-MIB::jnxOperatingDescr
+                        op: 'contains'
+                        value: 'Sensor'
```
Comment (ottorei) [inline includes/definitions/discovery/junos.yaml:23]: For some reason, on MX-80 it still discovers a "TFEB Intake Temperature Sensor" after applying this patch. Memory units are in GiB now.

Comment (ottorei) [reply includes/definitions/discovery/junos.yaml:23]: This "TFEB Intake Temperature Sensor" is discovered in CPU and Memory sections.

Debug capture shows this line on mempools: jnxOperatingDescr.* = "TFEB Intake temperature sensor"

Comment (ottorei) [reply includes/definitions/discovery/junos.yaml:23]: Also adding "TFEB" into the CPU and Memory skip values seems to achieve the desired outcome atleast with my devices.

Comment (TheMysteriousX) [reply includes/definitions/discovery/junos.yaml:23]: Looks like 'sensor' is lower cased in your example - I can add lowercase versions of each.

**Comment by ottorei** [comment]
Can you also add the 'TFEB' filter?
-------- Alkuperäinen viesti --------Lähettäjä: Adam Bishop <notifications@github.com> Päivämäärä: 20.1.2021  18.02  (GMT+02:00) Saaja: librenms/librenms <librenms@noreply.github.com> Kopio: ottorei <otto@digipahuus.fi>, Comment <comment@noreply.github.com> Aihe: Re: [librenms/librenms] Filter Juniper mempools (#12443) 
@TheMysteriousX commented on this pull request.



In includes/definitions/discovery/junos.yaml:
>                          op: '='
                         value: 0
+                    -
+                        oid: JUNIPER-MIB::jnxOperatingDescr
+                        op: 'contains'
+                        value: 'Fan'
+                    -
+                        oid: JUNIPER-MIB::jnxOperatingDescr
+                        op: 'contains'
+                        value: 'Sensor'

Looks like 'sensor' is lower cased in your example - I can add lowercase versions of each.

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/librenms/librenms/pull/12443#discussion_r561080425",
"url": "https://github.com/librenms/librenms/pull/12443#discussion_r561080425",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by TheMysteriousX** [comment]
I'm reluctant to filter TFEB as in other configurations it could have valid values exposed somewhere.

Filtering fan/sensor case insensitively should be sufficient - if you try the change I've added, you should see the erronious CPUs and mempools disappear.

**Review by murrant — APPROVED** [review]

**Comment by TheMysteriousX** [comment]
Refreshed the test data with the CPU change, hopefully checks should pass.

**Review by murrant — APPROVED** [review]
LGTM, thanks

**Pull request merged**